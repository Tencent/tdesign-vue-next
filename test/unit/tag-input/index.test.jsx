import { mount } from '@vue/test-utils';
import { vi } from 'vitest';
import TagInput from '@/src/tag-input/index.ts';

describe('TagInput', () => {
  describe(':props', () => {
    const tags = ['a', 'b', 'c', 'd'];
    it(':basic', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':label', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} label="Controlled: " />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':max', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} max={3} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':collapsed', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} min-collapsed-num={1} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':status:disabled', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} disabled />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':status:readonly', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} readonly />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':status:success', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} status={'success'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':status:warning', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} status={'warning'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':status:error', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} status={'error'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':draggable', () => {
      const wrapper = mount({
        render() {
          return <TagInput value={tags} dragSort />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('@event', async () => {
    it('blur', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <TagInput onBlur={fn} />;
        },
      });
      await wrapper.find('input').trigger('blur');
      expect(fn).toHaveBeenCalledTimes(1);
    });

    it('change', async () => {});

    it('clear', () => {});

    it('drag-sort', () => {});

    it('enter', async () => {
      // const fn = vi.fn();
      // const wrapper = mount({
      //   render() {
      //     return <TagInput onEnter={fn} />;
      //   },
      // });
      // const el = wrapper.find('input');
      // await el.trigger('keydown.enter');
      // expect(fn).toHaveBeenCalledTimes(0);
      // await el.setValue('a');
      // expect(el.element.value).toBe('a');
      // await el.trigger('keydown.enter');
      // expect(el.element.value).toBe('');
      // expect(fn).toHaveBeenCalledTimes(1);
    });

    it('input-change', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <TagInput onInputChange={fn} />;
        },
      });
      await wrapper.find('.t-input__inner').setValue('a');
      expect(fn).toHaveBeenCalledTimes(1);
    });

    it('mouse-enter', () => {
      // const fn = vi.fn();
      // const wrapper = mount({
      //   render() {
      //     return <TagInput onMouseenter={fn} />;
      //   },
      // });
      // const el = wrapper.find('.t-input__inner');
      // el.trigger('mouseenter');
      // expect(fn).toHaveBeenCalledTimes(1);
    });

    it('mouse-leave', () => {});

    it('paste', () => {});

    it('remove', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <TagInput value={['a']} onRemove={fn} />;
        },
      });
      const el = wrapper.findAll('.t-icon')[0];
      el.trigger('click');
      expect(fn).toHaveBeenCalledTimes(1);
    });
  });

  describe('<slot>', () => {
    it('<collapsedTags>', () => {
      const wrapper = mount(TagInput, {
        scopedSlots: {
          collapsedItems: '<div></div>',
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
