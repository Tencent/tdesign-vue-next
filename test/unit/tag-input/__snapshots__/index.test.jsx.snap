// Vitest Snapshot v1

exports[`TagInput > :props > :basic 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :collapsed 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        +
        3
        
        <!---->
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :draggable 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
        draggable="true"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
        draggable="true"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
        draggable="true"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
        draggable="true"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :label 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <div
        class="t-tag-input__prefix"
      >
        Controlled: 
      </div>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :max 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :status:disabled 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-disabled t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--disabled t-size-m"
      >
        <!---->
        
        a
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--disabled t-size-m"
      >
        <!---->
        
        b
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--disabled t-size-m"
      >
        <!---->
        
        c
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--disabled t-size-m"
      >
        <!---->
        
        d
        
        <!---->
      </span>
      
    </div>
    <input
      class="t-input__inner"
      disabled=""
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :status:error 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-error t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :status:readonly 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-is-readonly t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        a
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        b
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        c
        
        <!---->
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        d
        
        <!---->
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      readonly=""
      type="text"
      unselectable="on"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :status:success 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-success t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > :props > :status:warning 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-warning t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        a
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        b
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        c
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      <span
        class="t-tag t-tag--default t-tag--dark t-tag--close t-size-m"
      >
        <!---->
        
        d
        
        <svg
          class="t-icon t-icon-close t-tag__icon-close"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
      
    </div>
    <input
      class="t-input__inner"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;

exports[`TagInput > <slot> > <collapsedTags> 1`] = `
<div
  class="t-input__wrap t-tag-input t-tag-input--break-line"
>
  <div
    class="t-input t-is-default t-input--prefix"
  >
    <!---->
    <div
      class="t-input__prefix"
    >
      
      
    </div>
    <input
      class="t-input__inner"
      placeholder="请输入"
      type="text"
    />
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
`;
